crates/apollo-compiler/src/database/def.rs

```diff
@@ -327,7 +327,10 @@ fn operation_definition(
     // check if there are already operations
     // if there are operations, they must have names
     // if there are no names, an error must be raised that all operations must have a name
-    let name = op_def.name().map(|name| name.text().to_string());
+    let name = op_def.name().map(|name| Name {
```

Comment:
many of these look like they could use the `name()` function instead of manually creating Name nodes each time. or is something preventing that?

Reply:
these are all a bit of special cases, the name() method we have takes an option of name and creates a node, some of these are option of named type, or something else. butttt i can just make a method for the hir node creation